File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        selfDivNums = []
        for i in range(left, right + 1):
            n = i
            ok = True
            while n > 0:
                digit = n % 10
                if digit == 0 or i % digit != 0:
                    ok = False
                    break
                n //= 10
            if ok:
                selfDivNums.append(i)
                
        return selfDivNums

File: 728-self-dividing-numbers/self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> selfDivNums = new ArrayList<>();
        for (int i = left; i <= right; i++){
            int n = i;
            boolean ok = true;
            while (n > 0){
                int digit = n % 10;
                if (digit == 0 || i % digit != 0){
                    ok = false;
                    break;
                } 
                n /= 10;
            }
            if (ok) selfDivNums.add(i);
        }

        return selfDivNums;
    }
}
